### DISCUSSION TITLE
Native Scout Support

### DISCUSSION THREAD

**Discussion by ZacharyDuBois** [filamentphp/filament#2755] [Ideas]
Hi,

Something I've noticed is Filament's global search (and all searches for that matter) create some fairly ugly/unoptimized queries. It would be a phenomenal addition to add native Laravel Scout support. It appears there was some efforts put in to attempt to support this with PR #1605.

Is this on the roadmap at all? It would significantly speed up the searches and filters.

Thanks for the awesome admin panel!

**Comment by danharrin** [comment]
It's possible to add Scout support to any search, but we haven't made this the default as I don't want to add another setup step.

What are you stuck on? I thought we documented it all.

**Comment by ZacharyDuBois** [comment]
The only documentation I was able to find was for [Scout with Filament tables](https://filamentphp.com/docs/2.x/tables/getting-started#searching-records-with-laravel-scout). I did not find anything other than that commit for Scout support for the global search.

**Reply by danharrin** [reply]
cc/ @ryangjchandler if you have a ScoutGlobalSearchProvider somewhere, could you share it in a Gist or something?

**Reply by ZacharyDuBois** [reply]
@ryangjchandler any update on this?

**Comment by dev3k** [comment]
sorry for replying to old discussion as I didn't find any solution, anyway I found a way with filament v3 , in resource page you can override getGlobalSearchResults like below, not sure if its the optimal solution but it works :) 

```php
public static function getGlobalSearchResults(string $search): Collection
    {
        return Product::search($search)
            ->take(static::getGlobalSearchResultsLimit())
            ->get()
            ->map(function (Model $record): ?GlobalSearchResult {
                $url = static::getGlobalSearchResultUrl($record);

                if (blank($url)) {
                    return null;
                }

                return new GlobalSearchResult(
                    title: static::getGlobalSearchResultTitle($record),
                    url: $url,
                    details: static::getGlobalSearchResultDetails($record),
                    actions: static::getGlobalSearchResultActions($record),
                );
            })
            ->filter();
    }
```

**Comment by coolsam726** [comment]
See the Docs: https://filamentphp.com/docs/3.x/tables/advanced#searching-records-with-laravel-scout

**Comment by ZacharyDuBois** [comment]
@coolsam726 - That is for table search, not global.
